src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+document.querySelectorAll('thead th').forEach((th) => {
+  th.addEventListener('click', () => {
+    const cellIndex = th.cellIndex;
+    const tableBody = document.querySelector('table tbody');
+    const rows = Array.from(tableBody.rows);
+
+    rows.sort((row1, row2) => {
+      let val1 = row1.cells[cellIndex].textContent.trim();
+      let val2 = row2.cells[cellIndex].textContent.trim();
+
+      if (
+        !isNaN(val1.replace(/[$,]/g, '')) &&
+        !isNaN(val2.replace(/[$,]/g, ''))) {
```

Comment:
The condition checks if both values are numbers by removing '$' and ',' characters. Ensure that the input data is consistent with this format, or consider handling other numeric formats if necessary.